lib/node/nodes/georeference-postal-code.js

```diff
@@ -1,33 +1,74 @@
 'use strict';
 
 var Node = require('../node');
+var debug = require('../../util/debug')('analysis:georeference-postal-code');
 
 var TYPE = 'georeference-postal-code';
 var PARAMS = {
     source: Node.PARAM.NODE(Node.GEOMETRY.ANY),
+    output_geometry_type: Node.PARAM.NULLABLE(
+        Node.PARAM.ENUM(Node.GEOMETRY.POINT, Node.GEOMETRY.POLYGON),
+        Node.GEOMETRY.POLYGON
+    ),
     postal_code: Node.PARAM.STRING(),
-    country: Node.PARAM.STRING()
+    country: Node.PARAM.NULLABLE(Node.PARAM.STRING()),
+    country_column: Node.PARAM.NULLABLE(Node.PARAM.STRING())
 };
 
-var GeoreferencePostalCode = Node.create(TYPE, PARAMS, { cache: true });
+var GeoreferencePostalCode = Node.create(TYPE, PARAMS, {
+    cache: true,
+    beforeCreate: function (node) {
+        if (!node.country && !node.country_column) {
+            throw new Error('You must indicate "country" or "country_column" param');
+        }
+    }
+});
 
 module.exports = GeoreferencePostalCode;
 module.exports.TYPE = TYPE;
 module.exports.PARAMS = PARAMS;
 
 GeoreferencePostalCode.prototype.sql = function() {
-    return queryTemplate({
+    var sql = queryTemplate({
         source: this.source.getQuery(),
         columns: this.source.getColumns(true).join(', '),
+        geocoder_function: getGecoderFunction(this.output_geometry_type),
         postal_code: this.postal_code,
-        country: this.country
+        country: this.getFunctionParam('country')
     });
+
+    debug(sql);
+
+    return sql;
 };
 
+GeoreferencePostalCode.prototype.getFunctionParam = function (name) {
+    if (this[name + '_column']) {
+        return this[name + '_column'];
+    }
+
+    if (this[name]) {
+        return '\'' + this[name] + '\'';
+    }
+
+    return 'NULL';
+};
+
+function getGecoderFunction(geometryType) {
+    switch (geometryType){
+        case Node.GEOMETRY.POLYGON:
+            return 'cdb_geocode_postalcode_polygon';
+        case Node.GEOMETRY.POINT:
+            return 'cdb_google_geocode_street_point';
```

Comment:
Ooops, a wild address function appeared! It should be `cdb_geocode_postalcode_point` instead.